apps/files/helpers.py

```diff
@@ -335,30 +342,143 @@ def get_url(self, short):
                        args=[self.left.file.id, self.right.file.id,
                              'file', short])
 
-    def get_files(self):
+    _files = None
+
+    @property
+    def files(self):
         """
         Get the files from the primary and:
         - remap any diffable ones to the compare url as opposed to the other
         - highlight any diffs
         """
-        left_files = self.left.get_files()
-        right_files = self.right.get_files()
-        different = []
-        for key, file in left_files.items():
-            file['url'] = self.get_url(file['short'])
-            diff = file['md5'] != right_files.get(key, {}).get('md5')
-            file['diff'] = diff
-            if diff:
-                different.append(file)
-
-        # Now mark every directory above each different file as different.
-        for diff in different:
-            for depth in range(diff['depth']):
-                key = '/'.join(diff['short'].split('/')[:depth + 1])
-                if key in left_files:
-                    left_files[key]['diff'] = True
-
-        return left_files
+        if not self._files:
+            self.right.files = self._remap_filenames()
+
+            left_files = self.left.files
+            right_files = self.right.files
+            different = []
+            for key, file in left_files.items():
+                file['url'] = self.get_url(file['short'])
+                diff = file['md5'] != right_files.get(key, {}).get('md5')
+                file['diff'] = diff
+                if diff:
+                    different.append(file)
+
+            # Now mark every directory above each different file as different.
+            for diff in different:
+                for depth in range(diff['depth']):
+                    key = '/'.join(diff['short'].split('/')[:depth + 1])
+                    if key in left_files:
+                        left_files[key]['diff'] = True
+
+            self._files = left_files
+        return self._files
+
+    @memoize(prefix='file-viewer-remap-files', time=60 * 60)
+    def _remap_filenames(self):
+        """
+        Remap the filenames of add-ons whose directory contents have
+        been re-arranged. Currently handles SDK add-ons moving from
+        the `cfx` packaging format to the `jpm` format.
```

Comment:
Having a detailed list of path moves in the dosctring would also help understand the rest of the method code.